src/main/scala/org/constellation/datastore/SnapshotTrigger.scala

```diff
@@ -49,6 +54,11 @@ class SnapshotTrigger(periodSeconds: Int = 5)(implicit dao: DAO, cluster: Cluste
       IO.unit
     )
 
+  def handleError(err: SnapshotError, stateSet: SetStateResult) =
+    dao.cluster.compareAndSet(Set(NodeState.SnapshotCreation), stateSet.oldState, skipBroadcast = true) >>
```

Comment:
You have a cluster dependency here, can you use it instead of using `dao`?

Reply:
It was purely copied from previous implementation, fixed.